File: integrations/yasa/YASA-Engine-sembridge/test/callchain/js/test_callchain.js
const { exec } = require('child_process');
const mysql = require('mysql');

/**
 * Test SQL injection sink detection
 */
function testSqlInjection(userInput) {
    const connection = mysql.createConnection({
        host: 'localhost',
        user: 'root',
        password: 'password',
        database: 'test'
    });

    connection.connect();

    // This should be detected as a sink match
    const query = "SELECT * FROM users WHERE name = '" + userInput + "'";
    connection.query(query, function (error, results, fields) {
        if (error) throw error;
        console.log(results);
    });

    connection.end();
}

/**
 * Test command injection sink detection
 */
function testCommandInjection(command) {
    // This should be detected as a sink match
    exec(command, (error, stdout, stderr) => {
        if (error) {
            console.error(`exec error: ${error}`);
            return;
        }
        console.log(`stdout: ${stdout}`);
        console.error(`stderr: ${stderr}`);
    });
}

/**
 * Test eval sink detection
 */
function testEvalInjection(code) {
    // This should be detected as a sink match
    eval(code);
}

// Main execution
testSqlInjection('admin');
testCommandInjection('ls -la');
testEvalInjection('console.log("test")');

File: integrations/yasa/YASA-Engine-sembridge/test/callchain/go/test_callchain.go
package main

import (
	"database/sql"
	"fmt"
	"os/exec"
)

// TestSqlInjection tests SQL injection sink detection
func TestSqlInjection(userInput string) {
	db, _ := sql.Open("mysql", "user:password@/dbname")
	defer db.Close()

	// This should be detected as a sink match
	query := "SELECT * FROM users WHERE name = '" + userInput + "'"
	db.Query(query)
	db.Exec(query)
}

// TestCommandInjection tests command injection sink detection
func TestCommandInjection(command string) {
	// This should be detected as a sink match
	cmd := exec.Command("sh", "-c", command)
	output, err := cmd.Output()
	if err != nil {
		panic(err)
	}
	fmt.Println(string(output))
}

func main() {
	TestSqlInjection("admin")
	TestCommandInjection("ls -la")
}
